src/main/java/sparksoniq/jsoniq/item/Item.java

```diff
@@ -163,31 +163,57 @@ public static boolean checkEquality(Item v1, Item v2) {
         return compareItems(v1, v2) == 0;
     }
 
-    public abstract List<Item> getItems() throws OperationNotSupportedException;
+    public List<Item> getItems() {
+        throw new RuntimeException("Item is not an array or object.");
```

Comment:
I think this error message should be changed to "Item is not an array" in order to not be misleading.
ObjectItem does not override getItems() method.